janitor/functions.py

```diff
@@ -1021,3 +1025,171 @@ def _round_to_fraction(number, denominator, digits=np.inf):
     df[colname] = df[colname].apply(_round_to_fraction_partial)
 
     return df
+
+
+@lru_cache(maxsize=32)
+def _convert_currency(
+    from_currency: str = None,
+    to_currency: str = None,
+    historical_date: date = None,
+):
+    """
+    Currency conversion for Pandas DataFrame column.
+    Helper function for `convert_currency` method.
+    The API used is: https://exchangeratesapi.io/
+    """
+
+    url = "https://api.exchangeratesapi.io"
+
+    if historical_date:
+        assert isinstance(historical_date, date) | isinstance(
+            historical_date, datetime
+        ), "historical_date must be a date or datetime object!"
+        if isinstance(historical_date, datetime):
+            assert historical_date >= datetime(
```

Comment:
I'd encourage you to change this to:
```suggestion
            if historical_date < datetime(1999, 1, 4):
                raise ValueError("datetime must be later than 1999-01-04!")
```

For the same reasons as above.